packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -313,6 +315,32 @@ void main() {
       });
     });
   });
+
+    testUsingContext(
+    'runner disable telemetry with flag',
+    () async {
+      io.setExitFunctionForTests((int exitCode) {});
+      fileSystem = MemoryFileSystem.test();
```

Comment:
Shouldn't this be set before the test runs?

Reply:
Yes, but the setup in this file was only for the other tests which expected errors. There was no global setup being run under main

Reply:
Actually, the rest of this test never references `fileSystem`, so you can just instantiate it in the anonymous closure on line 338 (which will be 337 after you delete this line)